Testcase for simultaneous keypresses

gedankenexperimenter opened the pull request:
This test runs two keypresses in the same cycle, and checks to see that only one report is generated, and that it contains both active keycodes (and no others).

This test currently fails; see #941.

obra commented:
https://github.com/keyboardio/Kaleidoscope/commit/dbfefd463052c205e58fa4d186b1c0e843ec68e6 is the equivalent test I'd worked up a few hours ago.

gedankenexperimenter commented:
That test is almost equivalent; it doesn't verify that there are no unexpected keycodes in the report(s). I wanted to do it this way to put an example on record.

gedankenexperimenter commented:
I reworked this testcase, now that @obra has solved the mystery of the multiple reports, and corrected my understanding of what's going on with the simulator. I added a third key in the same cycle, and as expected, only two reports were generated, not three. This is consistent with what the HID code does to deal with the problem, and not, in my opinion, a bug.

I changed the testcase so that it simply verifies that at least one report was sent, and checks that the contents of the last one are what was expected (the three keycodes for the pressed keys, and no others). It now also passes the test, rather than failing.

Closes #941.

Furthermore, we should add some testcases for the rollover cases involving modifier-flagged keys, but that's for another PR. Thanks to @obra for finding the problem when I was unable to navigate my way through the code.
